### PR TITLE
Switches Redis client to Lettuce

### PR DESCRIPTION
**Pull request by mzamani1**

### CODE REVIEW

File: src/main/java/io/dropwizard/redis/HostAndPort.java [hunk 1]
```diff
@@ -7,15 +7,15 @@
 
 import javax.validation.constraints.NotNull;
 
-public class Node {
+public class HostAndPort {
```
Comment (jplock) [inline src/main/java/io/dropwizard/redis/HostAndPort.java]: Guava’s HostAndPort?

Comment (mzamani1) [reply src/main/java/io/dropwizard/redis/HostAndPort.java]: Good point, wasn't sure if the Guava Jackson module supported serialization for that, but it does, so I'll switch.

**Comment by mzamani1** [comment]
TODO:
- [ ] Metrics integration
- [ ] Unit tests
- [ ] Integration tests
- [ ] Support for the remaining `ClientResources` stuff I left out for now 
- [ ] Implementation for the `RedisClientFactory`

**Comment by mzamani1** [comment]
Planning on finishing this up this weekend

**Comment by mzamani1** [comment]
@joschi @jplock @pstackle I think this is ready to be merged in, pending your reviews hopefully.

**Pull request merged**